### PR TITLE
Add selector-disallowed-list rule

### PR DESCRIPTION
**Pull request by doing-art**
> Which issue, if any, is this issue related to?

Have added a new `selector-disallowed-list` rule as discussed in issue [#5116](https://github.com/stylelint/stylelint/issues/5116) .

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by doing-art** [comment]
Hello everyone.

[One check](https://github.com/stylelint/stylelint/pull/5239/checks?check_run_id=2317461981) related to this PR fails. Should I fix it? Or it is an infrastructural issue and you will fix it on your own?

**Comment by hudochenkov** [comment]
@doing-art It was a weird issue. Tests restart helped.

**Comment by doing-art** [comment]
> @doing-art It was a weird issue. Tests restart helped.

Thanks a lot for your help. I really appreciate it

**Review by ybiquitous — COMMENTED** [review]
@doing-art Thanks for the contribution! LGTM for the code changed. 👍 

I've left some comments for the document, so please confirm them.

File: lib/rules/selector-disallowed-list/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    .foo > .bar
+/** ↑
+ * This is selector */
+```
+
+## Options
+
+`string | regex | array of string and regex`
+
+### Given the string:
```
[the remark this answers is not in the document]
Comment (doing-art) [reply lib/rules/selector-disallowed-list/README.md]: @ybiquitous Thanks a lot for sharing your vision. I couldn't agree more with your comments.

Updated the documentation as you suggested. Also adjusted all related tests to make them check cases enumerated in the updated docs.

Please, take a look at the PR when you have a chance.

File: lib/rules/selector-disallowed-list/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * This is selector */
+```
+
+## Options
+
+`array|string|regexp`: `["array", "of", "selectors", /or/, "/regex/"]|"selector"|"/regex/"`
+
+If a string is surrounded with `"/"` (e.g. `"/\.foo/"`), it is interpreted as a regular expression.
+
+### Given:
```
Comment (ybiquitous) [inline lib/rules/selector-disallowed-list/README.md:18]: [suggestion] Could you remove `###` like other rules' documents?

```suggestion
Given:
```

**Review by ybiquitous — APPROVED** [review]
@doing-art Thank you for addressing my suggestions!
I've left a trivial suggestion, but your change looks good to me! 😄

**Review by jeddy3 — APPROVED** [review]
Thank you, LGTM.

**Comment by jeddy3** [comment]
Changelog entry added:

> - Added: `selector-disallowed-list` rule ([#5239](https://github.com/stylelint/stylelint/pull/5239)).

**Pull request merged**